File: backend/models/reward.js
"use strict";
exports.__esModule = true;
exports.rewardSchema = void 0;
var mongoose_1 = require("mongoose");
exports.rewardSchema = new mongoose_1.Schema({
    image: { type: String, required: true },
    price: { type: Number, required: true },
    title: { type: String, required: true },
    description: { type: String, required: true }
});
exports["default"] = (0, mongoose_1.model)("reward", exports.rewardSchema);

File: flutter/lib/models/Reward.dart
class Reward {
  final String id;
  final int price;
  final String title;
  final String description;
  final String image;

  Reward({
    required this.id,
    required this.price,
    required this.title,
    required this.description,
    required this.image,
  });

  Reward.fromJson(Map<String, dynamic> json)
      : id = json["_id"],
        price = json["price"],
        title = json["title"],
        description = json["description"],
        image = json["image"];
}
